src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -623,14 +622,16 @@ protected virtual void ValidateNodeResult(ExecutionContext context, ExecutionNod
                         + $" Field: {node.Name}, Type: {nonNullType}.");
                 }
 
-                objectType = nonNullType.ResolvedType as IObjectGraphType;
+                fieldType = nonNullType.ResolvedType;
             }
 
             if (result == null)
             {
                 return;
             }
 
+            var objectType = fieldType as IObjectGraphType;
+
             if (fieldType is IAbstractGraphType abstractType)
```

Comment:
Previously this `fieldType` would not have been unwrapped by the above lines of code, and so would skip all these checks in this `if` block.

Reply:
So only `nonnull` is problem here, not `list` ?

Reply:
Hmm... I think I tested it and determined it was also a problem.  I rewrote this code a couple times and probably missed it on the rewrite.

Reply:
Added tests to verify validation occurs for lists